ui/desktop/src/utils/loadEnv.ts

```diff
@@ -0,0 +1,36 @@
+import { execSync } from 'child_process';
+import path from 'path';
+import log from './logger';
+
+export function loadZshEnv(isProduction: boolean = false): void {
+  // Only proceed if running on macOS and in production mode
+  if (process.platform !== 'darwin' || !isProduction) {
+    log.info(`Skipping zsh environment loading: ${
+      process.platform !== 'darwin' ? 'Not running on macOS' : 'Not in production mode'
+    }`);
+    return;
+  }
+
+  try {
+    // Execute zsh and source the zshrc file, then export all environment variables
```

Comment:
Is this because `goosed` will read the API key from an env var? Can we just do that one instead of all env vars? How hard would it be to handle `bash` as well?

Alternative idea: let the user enter an API key password style somewhere in the app. When we spawn `goosed`, we could provide that one env var when we spawn the process. Seems a bit simpler and also more user friendly than needing to set an env var for an app to work, which I am not used to as a pattern.

Reply:
yeah - it will need it - but not sure exactly what is needed, so this makes it the same, for now, as `npm start`  - but ideally it would be goosed doing it I think?

Reply:
@alexhancock also consider this temporary until we work out how wr really discover and use config